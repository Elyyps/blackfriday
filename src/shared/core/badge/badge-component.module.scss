.badge {
  top: 0;
  position: absolute;

  &-right {
    right: 0;
    button {
      padding: 0 0 10px 10px;
      border-radius: 0 0 0 100%;
    }
  }
  &-left {
    left: 0;
    button {
      padding: 0 10px 10px 0;
      border-radius: 0 0 100% 0;
    }
  }

  &__dropdown {
    margin-top: 0;
    background-color: white;
    box-shadow: 0 0 8px 0 rgba(0, 0, 0, 0.18);
  }
  button {
    line-height: 0;
    font-size: 0;
    width: 56px;
    border: none;
    height: 56px;
    background: none;
    overflow: hidden;
    position: relative;
    background: $tint-gray-light;

    display: flex;
    justify-content: center;
    transition: transform 0.2s;
    &:focus {
      outline: 0;
    }
    &:hover {
      cursor: pointer;
      svg {
        transform: scale(1.2);
      }
    }
  }
}
