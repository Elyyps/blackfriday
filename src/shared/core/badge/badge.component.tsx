import * as React from "react";
import styles from "./badge-component.module.scss";
import { IconComponent } from "../icon";

interface IBadgeProps {
  children?: any;
  icon: string;
  position: "right" | "left";
}

const Badge = (props: IBadgeProps) => (
  <div className={`${styles["badge"]} ${styles[`badge-${props.position}`]}`}>
    <button>{props.icon ? <IconComponent icon={props.icon} size="18px" /> : ""}</button>
    <div className={styles["badge__dropdown"]} data-uk-dropdown="mode: click">
      {props.children}
    </div>
  </div>
);

export { Badge };
