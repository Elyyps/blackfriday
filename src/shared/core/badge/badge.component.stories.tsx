import * as React from "react";
import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";

import { Badge } from "./badge.component";

import ShareSVG from "@assets/icons/share.svg";
const notes = require("./badge.md");

storiesOf("Badge", module)
  .addDecorator(withA11y)
  .add("Basic implementation", () => <Badge position={"right"} icon={ShareSVG} />, {
    notes
  });
