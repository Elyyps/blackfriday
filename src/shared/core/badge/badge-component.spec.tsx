import React from "react";
import { shallow } from "enzyme";
import { Badge } from "./badge.component";
import ShareSVG from "@assets/icons/share.svg";
describe("[Badge]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(<Badge position={"right"} icon={ShareSVG} />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
