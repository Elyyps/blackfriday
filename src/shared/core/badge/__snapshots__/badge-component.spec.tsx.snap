// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Badge] should render component without crashing 1`] = `
<div
  className="badge badge-right"
>
  <button>
    <IconComponent
      icon="share.svg"
      size="18px"
    />
  </button>
</div>
`;
